### PR TITLE
Barchart fetcher implementation

### PR DESCRIPTION
**Pull request by hatskier**

### CODE REVIEW

File: src/fetchers/barchart/index.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+const logger =
+  require("../../utils/logger")("fetchers/barchart") as Consola;
+
+const URL = "https://ondemand.websol.barchart.com/getQuote.json";
+
+const barchartFetcher: Fetcher = {
+  async fetchAll(symbols: string[], credentials): Promise<PriceDataFetched[]> {
+    // Fetching prices from Barchart
+    const response: any = await axios.get(URL, {
```
Comment (ppedziwiatr) [inline src/fetchers/barchart/index.ts:13]: I'm a fan of small (and having one purpose..) functions, so I would probably split this implementation into at least three smaller functions (this makes the whole fetching process much more readable, without getting too much into details):
```
const barchartResponse = fetchResponse(); // lines [11-21]
check(barchartResponse); // lines [23-36]
return buildPrices(barchartResponse); // lines [38-51]
```
- but that's just my preference...

Comment (jakub-wojciechowski) [reply src/fetchers/barchart/index.ts:13]: We've been discussing this while talking about fetchers refactor as the same comment could be made for any existing fetcher. Introducing the strategy pattern there could be useful in the context of a dynamic fetcher when we could set parameters from UI.

I've created an issue: https://app.clickup.com/t/m993kd

File: src/fetchers/barchart/index.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      params: {
+        apikey: credentials?.credentials.barchartApiKey,
+        symbols: symbols.map(convertSymbol).join(","),
+      },
+    });
+
+    // TODO: remove later when barchart fetcher will be 100% stable
+    logger.info("Barchart response: " + JSON.stringify(response.data));
```
Comment (jakub-wojciechowski) [inline src/fetchers/barchart/index.ts:21]: Could we also track the api response time - I'm wondering what could be the minimum timeout at our location?

Comment (hatskier) [reply src/fetchers/barchart/index.ts:21]: We already track it in performance tracker

**Pull request merged**